readme.txt
=== Woo Product Table - Free WooCommerce Table view solution ===
Contributors: codersaiful,codeastrology
Tags: wc product table,woocommerce product table, product table, wc table, quick order table, wholesale table, restaurants table
Requires at least: 4.0.0
Tested up to: 5.2.2
Requires PHP: 5.6
Stable tag: 4.3
License: GPLv2 or later
License URI: https://www.gnu.org/licenses/gpl-2.0.html

Product Table's pro Features in FREE. Easily display your products as table by tiny shortcode. Such: [Product_Table id='1']. Support default editor, any Page Builder & Gutenberg.

== Description ==
Auto Responsive Mobile & Tab, Nice Design, Fast Speed *WooCommerce Product Table*. Tiny Shortcode, Easy customizable, Ajax Add to cart, Ajax Search, Pagination & Load More.
*Woo Product Table - Free WooCommerce Table view solution* offers to show all products display as a table in page by light shortcode. Fully responsive and mobile friendly. Easily customizable - color,background,title,text color etc.
.

**WooCommerce Product Table** plugin offers to display your shop products in one page by shortcode as table. 

* Shop page as Table
* Available add to cart in Table
* Ajax add to cart
* Advanced SearchBox to search from all products
* Well adaptable with your theme design
* Easy customization from Dashboard
* YITH WishList Support in table
* Has Option to Hide Table Header 
* All text of your table are editable now.

Add the shortcode to any existing post or page:

`[Product_Table ID='123' name='Table Name']`
*ID* - will be post's ID(required).It will generate automatically when you create your table. 

*name* - only for identifying your table's shortcode.(Optional)


This plugin will help any body to show All product in 1 page as Table. Also able to change/customize. Go to: Dashboard->WPT Table . 
If anybody found any issue, Please inform me to wo-info@detatech.xyz.


*All custom field and taxonomy plugin supported
* More Light Weight
* Table configuration saved as Table ID
* Use Table  by Table ID
* Add Custom Taxonomy By a Click easily, no need to add in shortcode
* Add Custom Field By  a Click easily, no need to add in shortcode
* YITH quick view
* WooCommerce Product Addon supported
* No need shortcode editing
* Able to change and customize of table's style,column,content from custom_post type
* Variation wise LightBox image Display
* And So many many feature Update

== Installation ==


1. Upload the plugin files to the `/wp-content/plugins/woo-product-table` directory, or install the plugin through the WordPress plugins screen directly.
2. Activate the plugin through the 'Plugins' screen in WordPress
3. Use the Woo Table->Setting screen to configure the plugin's default setting.


== Frequently Asked Questions ==

= Why Woo Product Table? =

To see your all products as a table in a page by shortcode.User friendly interface with easy options.If you want to increase user engagement in your site then it's the best solution.

= What is default Shortcode? =

Default Shortcode is [Product_Table id='123' name='Home Table'] as well as also able to change. Go to Product Table page from dashboard.

= Can I set product limitation to show in one table? =

Yes. You can set the product limit to show in yout able. Eg. You have 100 products in your site then you can easily show 50 of them.

= How to use? =

Install and activate. Then go to ( Dashboard -> Product Table-> Add New ).
 Give a suitable name of your table. Configure your table according to your need and click publish. Finally, copy the shortcode and paste it to your desired page or post.
That's it. So easy, Right !!!

= How to show specific Category products ? =

You can easily show specific category products. While creating new table click on the basic tab. You will find an option to select your category. You can also choose multiple categories.


= Is it suitable for any theme ? =
 Yes. *WooCommerce Product Table* will adapt with your theme design.

== Screenshots ==

`/assets/screenshot-1.png`
`/assets/screenshot-2.png`
`/assets/screenshot-3.png`
`/assets/screenshot-4.png`
`/assets/screenshot-5.png`

== Changelog ==

= 2.0 =
* Auto Responsive for Mobile
* Auto Responsive for Tab
* More Clean Design
* Bug Fix

= 1.9 =
* Fixed - javascript conflict issue of other cart plugin 
* custom field supported
* custom taxonomy supported
* Undefine index issue fixed for free version
* Pagination is free now
* Column Move added in Free

= 1.8 =
* YITH Quick View Added at Free
* Attribute Collumn Added at Free

= 1.7 =
* Many pro feature in free now.
* Translation issue fixed
* Different Configuration available for different table shortcode
* Bug Fix
* SSL issue fix for style and js file

= 1.6 =
* Bug Fix
* Supported with latest WooCommerce
* Supported with latest WordPress

= 1.6 =
* Removed old Shortcode
* Awesome Footer Cart Added
* Per product cart count and will update globally
* Cart quantity will display with per [Add to cart] button - for added item
* YITH Quote request Supported
* YITH Wishlist Supported
* YITH Quick View Supported
* Adding spin for loading - add to cart.
* Quote Request issue fix,
* js issue fixed,
* All text - Translateable 
* Old shortcode's feature has totally removed
* Added and Adding text removed from basic tab

= 1.5 =
* Easy shortcode System
* Table Manage by only one ID.
* Bug Fix

= 1.4 =
* Bug Fix
* Mobile Responsive
* Configure Page
* Column Move
* Ajax Add to cart
* So many pro feature in Free Version

= 1.3 =
* Shortcode Supporting in Custom Field.
* Default value issue fixed for Variation of product.
* Variations/Attribute in different column issue fixed.
* Now able to add Attributes as Individual Column.
* Popup Notice Enable/Disable Feature added.
* Cart Validation Message
* Compatible with all Min Max Plugin
* Removed default quantity from configuration page
* Code Optimized for better performance and speed.
* Load More button show/hide option
* Speed fast
* Advance search box's added default search order and order_by based on shortcode
* date and modied_date collumn added at Table
* Configure Page's design changed - So smart/so cool
* "On Back Order" available now in Table
* Mini cart update
* Advance Cart Update
* send 'add_to_cart','added_to_cart' event
* Configuration's value Reset option fix when plugin update
* Filter Text change option  of Filter
* Reset Button Text change option of Filter
* Thumbs image size issue fix
* Mini-cart Update automatically
* Fix Responsive Issue for TD's height
* Fix Responsive Issue for TD's width

= 1.2 =
* Bug Fix
* Stable with latest WooCommerce
* Stable with latest WordPress
* Ajax Add to cart
* Template for Table
* Quantity Bug fix

= 1.1 =
* Fix issue for no_woocommerce_fact.

= 1.0 =
* Just Start First version.

changelog.txt
== Changelog ==
5.2
Auto Responisve for Mobile, Tab, Desktop
Faster speed
Load More issue fix on new version
taxonomy, Custom field issue fixed
Minicart Load Alert Remove
column remove from tab option added
Drescription column removed and added in Product Title column
Cart meta issue, Duplicate cart item issue fixed

5.1
Admin tab issue fix
other plugin's conflict issue fixed
javascript other cart plugin issue fixed

5.0
Translation issue fixed
Different Configuration available for different table shortcode
Bug Fix
SSL issue fix for style and js file
Individual configurations Bug Fix
Sku,price,custom_metafiled wise ordering fix
And so on...

4.9
Set custom Category/Taxonomy in Search box with custom Order
Post_Include feature added. Able to set specific product in table with custom order Easily.

4.8
Yoast SEO conflict Remove
Custom-field spelling fix
Mini-Filter fixed, Now only show available Taxonomy
Advance Search box option change to select tag and customizeable

4.7
Footer Cart Customization features

4.6
Awesome Footer Cart Added
Per product cart count and will update globally
Cart quantity will display with per [Add to cart] button - for added item
Cart quantity will display with per product - for addedToCartSelected button
Adding spin for loading - add to cart.
Quote Request issue fix,
js issue fixed,
All text - Translateable 
'Products not founded' message is changeable from configure
Old shortcode's feature has totally removed
Added and Adding remvoe from basic tab

4.5
product_cat_ids attributes supported in new shortcode.

4.4
Table Design customizer Added
Table's all element is custimzeable


=4.3 (01/29/2019) =
Pagination On/Off feature added

=4.2 (01/24/2019) =
Table head Hiding Mobile's CSS issue Fix
AddToCart's vertical middle issue fixed for Responisve version
MiniCart's None Issue fixed
TempNumber updated - Now anybody can use any post duplicator plugin.

=4.1 (01/22/2019)=
More Ligthweight
Table configuration saved as Table ID
Use Table  by Table ID
Edit any table from Table ID
Lessen The Shortcode Length
Enable to Hide Table Header
Add Custom Taxonomy By a Click easily, no need to add in shortcode
Add Custom Field By  a Click easily, no need to add in shortcode
No need shortcode editing
Able to change and customize of table's style,colum,content from custom_post type
But Fix
Variation wise LightBox image Display
And So many many feature Update

= 4.0 (01/18/2019)=
Bug Fix
Attribute Display in another column
Variations in different column issue fixed.
Light Updated
Full Size image display in Light Box 
Return to default quantity after added to cart
Uncheck automatically after added to cart

= 3.9 (11/28/2018)=
Shortcode Supporting in Custom Field.
Default value issue fixed for Variation of product.
Variations in different column issue fixed.
Now able to add Attributes as Individual Column.
Popup Notice Enable/Disable Feature added.

= 3.8 (11/27/2018)=
Cart Validation Message
Compatible with all Min Max Plugin
Removed default quantity from configuration page
Code Optimized for better performance and speed.
Load More button show/hide option

= 3.7 (11/21/2018)=
Advance search box's added default search order and order_by based on shortcode
date and modied_date collumn added at Table
Configure Page's design changed - So smart/so cool
"On Back Order" available now in Table
Mini cart update
Advance Cart Update
send 'add_to_cart','added_to_cart' event


= 3.6 (10/29/2018)=
Configuration's value Reset option fix when plugin update
Filter Text change option  of Filter
Reset Button Text change option of Filter
Thumbs image size issue fix
Mini-cart Update automatically
Fix Responsive Issue for TD's height
Fix Responsive Issue for TD's width
product_description issue fixed

= 3.5 (10/17/2018)=
Bug Fix for CSS add to cart button in shop page
Bug fix for Instant search's result all check
Hide Load_More Button when all "unlimited product"
Load More button's bug fix, when click Search of "Advance Search"

= 3.4 (10/11/2018)=
Table design is updated based on new Design Trend.
Product Table Sort by SKU and also able to sort by any type Custom Meta value
Suppose you will able to sort by Custom Field Value
Able to sort by Custom Taxonomy
Front-End sorting will work properly based on Price
New Table template added named "Beautiful Blacky"
Documentation Updated based on new feature
New and Modern design for Documentation
Design is Updated for Our plugin's Dashboard part

= 3.3 =
Search Box's Spelling fix.
Search Box's all text are changeable from Configure Page.
[Type your Message] Placeholder is changeable from configure page
Variations issue solved for any type variations.
Table Stock column's "In Stock" and "Out of Stock" is now change able from configure page.
Mini Cart's "Cart","View Cart" and "Checkout" Text also change able from configure page
Bug fix

= 3.2 =
Categories, tags unlink bug fix, when click Search Button
Ajax Action Bug fix, when Search and Load button clicked
Custom Icon for Add to cart button. Such: Only Icon, No icon, with left icon, with right icon

= 3.1 =
YITH Quick View supported.
Product displaying from publish, added post_status => 'publish'
Added Custom Message at Configure page for "There is no product based on current Query."
Setting product_single page link for external and grouped product
Bug Fix for URL of variations product when "no_ajax_action"
Link remove for Categories and Tags from Configure page
[Quick Buy] Feature Added for Each Product. See in configure page "Enable Quick Buy Button" and set it as Yes to Activate.
Image/Thumbs lighBox Enable Disable Features Added
Mobile Responsive issue fix of "smart thin" table Template
Instance Search Issue Fixed. Now "Instance Search" is case insensitive.
Search Box's item sorting. Able to control from Configure page.


= 3.0 =
Search Button's Bug Fix
Instance Search's Bug fix
Live column Sorting ascending/descending
Cart Removing Feature at Mini-Cart
Able to change "Items/Item" text from configure

includes/functions.php
<?php

/**
 * getting Config value. If get config value from post, then it will receive from post, Otherwise, will take data from Configuration value.
 * 
 * @param type $table_ID Mainly post ID of wpt_product_table. That means: its post id of product table
 * @return type Array
 */
function wpt_get_config_value( $table_ID ){
    $config_value = $temp_config_value = get_option( 'wpt_configure_options' );
    $config = get_post_meta( $table_ID, 'config', true );
    if( !empty( $config ) && is_array( $config ) ){
        $config_value = $config;
    }
    $config_value['footer_cart'] = $temp_config_value['footer_cart'];
    $config_value['footer_cart_size'] = $temp_config_value['footer_cart_size'];
    $config_value['footer_possition'] = $temp_config_value['footer_possition'];
    $config_value['footer_bg_color'] = $temp_config_value['footer_bg_color'];
    $config_value['thumbs_lightbox'] = $temp_config_value['thumbs_lightbox'];
    $config_value['disable_cat_tag_link'] = $temp_config_value['disable_cat_tag_link'];
    return $config_value;
}
// Hook into Woocommerce when adding a product to the cart

add_filter( 'woocommerce_add_to_cart_fragments', 'wpt_per_item_fragment', 999 , 1 );

if( !function_exists( 'wpt_per_item_fragment' ) ) {
	function wpt_per_item_fragment($fragments)
	{
		ob_start();
                $Cart = WC()->cart->cart_contents;
                $product_response = false;
                if( is_array( $Cart ) && count( $Cart ) > 0 ){
                    foreach($Cart as $perItem){
                        //var_dump($perItem);
                        $pr_id = (String) $perItem['product_id'];
                        $pr_value = (String) $perItem['quantity'];
                        $product_response[$pr_id] = (String)  (isset( $product_response[$pr_id] ) ? $product_response[$pr_id] + $pr_value : $pr_value);
                        //$fragments["span.wpt_ccount.wpt_ccount_$pr_id"] = "<span class='wpt_ccount wpt_ccount_$pr_id'>$pr_value</span>";
                    }
                }
                //$fragments["span.wpt_ccount"] = "";
                if( is_array( $product_response ) && count( $product_response ) > 0 ){
                    foreach( $product_response as $key=>$value ){
                        //var_dump($perItem);
                        $pr_id = (String) $key;
                        $pr_value = (String) $value;
                        $fragments["span.wpt_ccount.wpt_ccount_$pr_id"] = "<span class='wpt_ccount wpt_ccount_$pr_id'>$pr_value</span>";
                    }
                }
                $fragments['.wpt-footer-cart-wrapper>a'] = '<a href="' . wc_get_cart_url() . '">' . WC()->cart->get_cart_subtotal() . '</a>';
		echo wp_json_encode($product_response);
		
		$fragments["wpt_per_product"] = ob_get_clean();
                //WC_AJAX::get_refreshed_fragments();
		return $fragments;
	}
}

/**
 * Generate paginated links based on Args.
 * 
 * @param type $args Args of WP_Query's
 * @return type String
 */
function wpt_paginate_links( $args = false ){
    $html = false;
    if( $args ){
        $product_loop = new WP_Query($args);
        $big = 99999999;
        $paginate = paginate_links( array(
            'base' => str_replace( $big, '%#%', esc_url( get_pagenum_link( $big ) ) ),
            'format' => '?paged=%#%',
            'mid_size'  =>  3,
            'prev_next' =>  false,
            'current' => max( 1, $args['paged'] ),
            'total' => $product_loop->max_num_pages
        ));
        $html .= $paginate; 
    }
    return $html;
}

/**
 * Generate full pagination based on Args.
 * 
 * @param type $args Args of WP_Query's
 * @return type String
 */
function wpt_pagination_by_args( $args = false, $temp_number = false ){
    $html = false;
    if( $args ){
        $html .= "<div class='wpt_table_pagination' data-temp_number='{$temp_number}'>";
        $paginate = wpt_paginate_links( $args );
        $html .= $paginate; 
        $html .= "</div>";
    }
    return $html;
}
/**
 * Generate Product's Attribute
 * 
 * @global type $product Default global product variable, it will only work inside loop
 * @param type $attributes Array
 * @return string 
 */
function wpt_additions_data_attribute( $attributes = false ){
    global $product;
    $html = false;
    if( $attributes && is_array( $attributes ) && count( $attributes ) > 0 ){
        foreach ( $attributes as $attribute ) :
        $html .= "<div class='wpt_each_attribute_wrapper'>";
            $html .= "<label>" . wc_attribute_label( $attribute->get_name() ) . "</label>";
            
            $values = array();

            if ( $attribute->is_taxonomy() ) {
                    $attribute_taxonomy = $attribute->get_taxonomy_object();
                    $attribute_values = wc_get_product_terms( $product->get_id(), $attribute->get_name(), array( 'fields' => 'all' ) );

                    foreach ( $attribute_values as $attribute_value ) {
                            $value_name = esc_html( $attribute_value->name );

                            if ( $attribute_taxonomy->attribute_public ) {
                                    $values[] = '<a href="' . esc_url( get_term_link( $attribute_value->term_id, $attribute->get_name() ) ) . '" rel="tag">' . $value_name . '</a>';
                            } else {
                                    $values[] = $value_name;
                            }
                    }
            } else {
                    $values = $attribute->get_options();

                    foreach ( $values as &$value ) {
                            $value = make_clickable( esc_html( $value ) );
                    }
            }

	$html .= apply_filters( 'woocommerce_attribute', wpautop( wptexturize( implode( ', ', $values ) ) ), $attribute, $values );
            
        $html .= '</div>';
        endforeach;
    }
    return $html;
}

/**
 * Checking Value for Select option tag
 * Used in shortcode.php file actually
 * 
 * @param type $got_value
 * @param type $this_value
 * @return type String
 */
function wpt_check_sortOrder( $got_value = false, $this_value = 'nothing' ){
    return $got_value == $this_value ? 'selected' : ''; 
}

/**
 * To get Final Columns List as Array, where will unavailable default disable_column
 * 
 * @return Array 
 */
function wpt_default_columns_array(){
    $column_array = WOO_Product_Table::$columns_array;
    /**
     * To this disable array, Only available keywords of Column Keyword Array
     * 
     */
    $disable_column_keyword = WOO_Product_Table::$colums_disable_array;
    foreach( $disable_column_keyword as $value ){
        unset( $column_array[$value] );
    }
    return $column_array;
}

/**
 * We used this function to get default keywords array from default columns array
 * 
 * @return Array Only Keys of Column Array
 * @since 3.6
 */
function wpt_default_columns_keys_array(){
    return array_keys( wpt_default_columns_array() );
}

/**
 * We used this function to get default values array from default columns array
 * 
 * @return Array Only values of Column Array
 * @since 3.6
 */
function wpt_default_columns_values_array(){
    return array_values( wpt_default_columns_array() );
}

/**
 * Taxonomy column generator
 * clue is: tax_
 * 
 * @param type $item_key
 * @return String
 */
function wpt_taxonomy_column_generator( $item_key ){
    $key = 'tax_';
    $len = strlen( $key );
    $check_key = substr( $item_key, 0, $len );
    if( $check_key == $key ){
        return $item_key;
    }
}

/**
 * Custom Fields column generator
 * clue is: cf_
 * 
 * @param type $item_key
 * @return String
 */
function wpt_customfileds_column_generator( $item_key ){
    $key = 'cf_';
    $len = strlen( $key );
    $check_key = substr( $item_key, 0, $len );
    if( $check_key == $key ){
        return $item_key;
    }
}

/**
 * Making new String/description based on word Limit.
 * 
 * @param String $string
 * @param Integer $word_limit
 * @return String
 */
function wpt_limit_words( $string = '', $word_limit = 10 ){
    $words = explode( " ",$string );
    
    $output = implode( " ",array_splice( $words,0,$word_limit ) );
    if( count( $words ) > $word_limit ){
       $output .= $output . '...'; 
    }
    return $output;
}

/**
 * Go generate as Array from 
 * 
 * @param Array $string Obviously should be an Array, Otherwise, it will generate false.
 * @param Array $default_array Actually if not fount a real String, and if we want to return and default value, than we can set here. 
 * @return Array This function will generate comman string to Array
 */
function wpt_explode_string_to_array( $string,$default_array = false ) {
    $final_array = false;
    if ( $string && is_string( $string ) ) {
        $string = rtrim( $string, ', ' );
        $final_array = explode( ',', $string );
    } else {
        if( is_array( $default_array ) ){
        $final_array = $default_array;
        }
    }
    return $final_array;
}

/**
 * Generate each row data for product table. This function will only use for once place.
 * I mean: in shortcode.php file normally.
 * But if anybody want to use any others where, you have to know about $table_column_keywords and $wpt_each_row
 * both should be Array, Although I didn't used condition for $wpt_each_row Array to this function. 
 * So used: based on your own risk.
 * 
 * @param Array $table_column_keywords
 * @param Array $wpt_each_row
 * @return String_Variable
 */
function wpt_generate_each_row_data($table_column_keywords = false, $wpt_each_row = false) {
    $final_row_data = false;
    if ( is_array( $table_column_keywords ) && count( $table_column_keywords ) > 0) {
        foreach ( $table_column_keywords as $each_keyword ) {
            $final_row_data .= ( isset( $wpt_each_row[$each_keyword] ) ? $wpt_each_row[$each_keyword] : false );
        }
    }
    return $final_row_data;
}

/**Generaed a Array for $wpt_permitted_td 
 * We will use this array to confirm display Table body's TD inside of Table
 * 
 * @since 1.0.4
 * @date 27/04/2018
 * @param Array $table_column_keywords
 * @return Array/False
 */
function wpt_define_permitted_td_array( $table_column_keywords = false ){
    
    $wpt_permitted_td = false;
    if( $table_column_keywords && is_array( $table_column_keywords ) && count( $table_column_keywords ) > 0 ){
        foreach( $table_column_keywords as $each_keyword ){
            $wpt_permitted_td[$each_keyword] = true;
        }
    }
    return $wpt_permitted_td;
}

/**
 * Generating <options>VAriation Atribute</option> for Product Variation
 * CAn be removed later.
 * 
 * @param type $current_single_attribute
 * @return string
 */
function wpt_array_to_option_atrribute( $current_single_attribute = false ){
    $html = '<option value>'.esc_html__( 'None', 'wpt_pro' ).'</option>';
    if( is_array( $current_single_attribute ) && count( $current_single_attribute ) ){
        foreach( $current_single_attribute as $wpt_pr_attributes ){
        $html .= "<option value='{$wpt_pr_attributes}'>" . ucwords( $wpt_pr_attributes ) . "</option>";
        }
    }
    return $html;
}

/**
 * For Variable product, Variation's attribute will generate to select tag
 * 
 * @param Array $attributes
 * @param Int $product_id
 * @param Int $temp_number
 * @return string HTML Select tag will generate from Attribute
 */
function wpt_variations_attribute_to_select( $attributes , $product_id = false, $default_attributes = false, $temp_number = false ){
    $html = false;
    $html .= "<div class='wpt_varition_section' data-product_id='{$product_id}'  data-temp_number='{$temp_number}'>";
    foreach( $attributes as $attribute_key_name=>$options ){

        $label = wc_attribute_label( $attribute_key_name );
        $attribute_name = wc_variation_attribute_name( $attribute_key_name );
        $only_attribute = str_replace( 'attribute_', '', $attribute_name);
        $default_value = !isset( $default_attributes[$only_attribute] ) ? false : $default_attributes[$only_attribute]; //Set in 3.9.0
        
		// Get an array of attributes
		$attr_array = get_the_terms( $product_id, $only_attribute);
        $html .= "<select data-product_id='{$product_id}' data-attribute_name='{$attribute_name}' placeholder='{$label}'>";
        $html .= "<option value='0'>" . $label . "</option>";
        foreach( $options as $option ){
			// Get the name of the current attribute
			$attr_name = $option;
			foreach( $attr_array as $attr ){
				if ( isset( $attr->slug ) && $option == $attr->slug) { $attr_name = $attr->name; }
			}
			
            $html .= "<option value='" . esc_attr( $option ) . "' " . ( $default_value == $option ? 'selected' : '' ) . ">" . $attr_name . "</option>";
        }
        $html .= "</select>";
        
    }
    $html .= "<div class='wpt_message wpt_message_{$product_id}'></div>";
    $html .= '</div>';

    return $html;
}



/**
 * Getting unit amount with unint sign. Suppose: Kg, inc, cm etc
 * woocommerce has default wp_options for weight,height etc's unit.
 * Example: for weight, woocommerce_weight_unit
 * 
 * @param string $target_unit Such as: weight, height, lenght, width
 * @param int $value Can be any number. It also can be floating point number. Normally decimal
 * @return string If get unit and value is gater than o, than it will generate string, otheriwse false
 */
function wpt_get_value_with_woocommerce_unit( $target_unit, $value ){
    $get_unit = get_option( 'woocommerce_' . $target_unit . '_unit' );
    return ( is_numeric( $value ) && $value > 0 ? $value . ' ' . $get_unit : false );
}

/**
 * Adding wpt_'s class at body tag, when Table will show.
 * Only for frontEnd
 * 
 * @global type $post
 * @global type $shortCodeText
 * @param type $class
 * @return string
 */
function wpt_adding_body_class( $class ) {

    global $post,$shortCodeText;

    if( isset($post->post_content) && has_shortcode( $post->post_content, $shortCodeText ) ) {
        $class[] = 'wpt_pro_table_body';
        $class[] = 'wpt_pro_table';
        $class[] = 'woocommerce';
    }
    return $class;
}
add_filter( 'body_class', 'wpt_adding_body_class' );


/**
 * Extend WordPress search to include custom fields
 *
 * Join posts and postmeta tables
 *
 * http://codex.wordpress.org/Plugin_API/Filter_Reference/posts_join
 */
function wpt_custom_search_join( $join ) {
    global $wpdb;
    $join .=' LEFT JOIN '.$wpdb->postmeta. ' ON '. $wpdb->posts . '.ID = ' . $wpdb->postmeta . '.post_id ';
    return $join;
}

/**
 * Modify the search query with posts_where
 *
 * http://codex.wordpress.org/Plugin_API/Filter_Reference/posts_where
 */
function wpt_custom_search_where( $where ) {
        global $pagenow, $wpdb;
        $myPreg = "(".$wpdb->posts.".post_title LIKE $1) OR (".$wpdb->postmeta.".meta_value LIKE $1";
        /*****************
        $myPreg .= " AND (";
        $myPreg .= $wpdb->postmeta.".meta_key='test'";


        $myPreg .= " OR ";
        $myPreg .= $wpdb->postmeta.".meta_key='_sku'";

        $myPreg .= ")";
        //***********************/
        $myPreg .= ")";
        $where = preg_replace("/\(\s*".$wpdb->posts.".post_title\s+LIKE\s*(\'[^\']+\')\s*\)/",$myPreg, $where );

    return $where;
}

/**
 * Prevent duplicates
 *
 * http://codex.wordpress.org/Plugin_API/Filter_Reference/posts_distinct
 */
function wpt_custom_search_distinct( $where ) {
        global $wpdb;
        return "DISTINCT";
        return $where;
    
}

$config_value = get_option( 'wpt_configure_options' );
if( isset( $config_value['sku_search'] ) && $config_value['sku_search'] == '1' ):
    add_filter('posts_join', 'wpt_custom_search_join' );
    add_filter( 'posts_where', 'wpt_custom_search_where' );
    add_filter( 'posts_distinct', 'wpt_custom_search_distinct' );
 endif;
